scripts/config/msu/text.nut

```diff
@@ -13,4 +13,39 @@
 	{
 		return this.color(::Const.UI.Color.NegativeValue, _string);
 	}
+
+	function colorizeValue( _value, _compareTo = 0, _invert = false, _addSign = true, _colorizeEqual = false )
+	{
+		if (_value < _compareTo)
+		{
+			if (!_addSign) _value *= -1;
+			return _invert ? this.colorGreen(_value) : this.colorRed(_value);
+		}
+
+		if (_value > _compareTo)
+		{
+			return (_addSign ? "+" : "") + (_invert ? this.colorRed(_value) : this.colorGreen(_value));
+		}
+
+		if (_value == _compareTo)
+		{
+			if (_colorizeEqual) return _invert ? this.colorRed(_value) : this.colorGreen(_value);
+			return _value;
+		}
+	}
+
+	function colorizeValueAndEqual( _value, _compareTo = 0, _invert = false, _addSign = true )
+	{
+		return this.colorizeValue(_value, _compareTo, _invert, _addSign, true);
+	}
+
+	function getChangeVerb( _value, _compareTo = 0, _invert = false, _upVerb = "increase", _downVerb = "decrease" )
+	{
+		if (_value >= _compareTo || _invert) return _upVerb : _downVerb;
+	}
+
+	function getIncreaseReduce( _value, _compareTo = 0, _invert = false )
+	{
+		return this.getChangeVerb(_value, _compareTo, _invert, "increase", "reduce");
+	}
```

Comment:
Are these actually useful? Just using a ternary in place seems like the same effort and you don't need to remember the functions.

Reply:
That's actually a good point, maybe this should color?

Reply:
> Are these actually useful? Just using a ternary in place seems like the same effort and you don't need to remember the functions.

Wdym?

Reply:
> That's actually a good point, maybe this should color?

You mean color the word?

Reply:
Yes

Reply:
> Wdym?

It is already simple to get such terms with a ternary. The simple case is:
val > 0 ? "increase" : "decrease" ; getChangeVerb(val)
But then you have cases where it's other than 0
val > 1 ? "increase" : "decrease" ; getChangeVerb(val, 1)
Or inverted
val < 0 ? "increase" : "decrease" ; getChangeVerb(val, 0, true)
Or different verbs
val < 0 ? "increase" : "reduce" ; getIncreaseReduce(val, 0, true)
Or the temp doesn't fit
val < 0 ? "increases" : "decreases" ; ???
Or different conditions
val != null && val = 0 ? "increases" : "decreases" ; ???

Basically I think it's barely worth, it's going to be more effort to remember what the function is called, and if you need a bunch of those you just make a local function

Reply:
I get your point now but just want to point out that the following is unnecessary:
> Or the temp doesn't fit
val > 0 ? "increases" : "decreases" ; getChangeVerb(val, 0, false, "increases", "decreases")

Because you can simply append `s` or `d` at the end of the returned verb when typing your tooltip to convert it into the tense you want.

So, in a string it will look something like:
```
"This is my tooltip string with an " + (val >= 0 ? "increase" : "decrease") + " of something something which leads to something something"
```

vs
```
"This is my tooltip string with an " + ::MSU.Text.getChangeVerb(val) + " of something something which leads to something something"
```

So, yeah I think the ternary is fine in this case.